src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -0,0 +1,89 @@
+package org.imperial.mrc.hint.db
+
+import org.jooq.tools.json.JSONArray
+import org.jooq.tools.json.JSONObject
+import org.springframework.stereotype.Component
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.imperial.mrc.hint.exceptions.CalibrateDataException
+import org.imperial.mrc.hint.models.CalibrateResultRow
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.Statement
+import java.util.Properties
+import java.util.Locale
+import java.sql.ResultSet
+import java.sql.SQLException
+
+const val DEFAULT_QUERY = """SELECT
+age_group,area_id,
+calendar_quarter,
+indicator,
+ROUND(lower, 4) AS lower,
+ROUND(mean, 4) AS mean,
+ROUND(mode, 4) AS mode,
+sex,
+ROUND(upper, 4) AS upper
+FROM data"""
+
+interface CalibrateDataRepository
+{
+    fun getDataFromPath(path: String): JSONArray
+}
+
+@Component
+class JooqCalibrateDataRepository: CalibrateDataRepository
+{
+
+    private fun convertToArrayList(resultSet: ResultSet): ArrayList<CalibrateResultRow> {
+        val list: ArrayList<CalibrateResultRow> = arrayListOf()
+        while (resultSet.next()) {
+            val row = CalibrateResultRow(
+                resultSet.getString("indicator"),
+                resultSet.getString("calendar_quarter"),
+                resultSet.getString("age_group"),
+                resultSet.getString("sex"),
+                resultSet.getString("area_id"),
+                resultSet.getFloat("mode"),
+                resultSet.getFloat("mean"),
+                resultSet.getFloat("lower"),
+                resultSet.getFloat("upper")
+            )
+            list.add(row)
+        }
+        return list
+    }
+
+    @Suppress("SwallowedException")
+    private fun getDataFromConnection(conn: Connection): ArrayList<CalibrateResultRow> {
+        try {
+            val query = DEFAULT_QUERY
+            val stmt: Statement = conn.createStatement()
+            val resultSet = stmt.executeQuery(query)
+            val arrayList = convertToArrayList(resultSet)
+            return arrayList
+        } catch (e: SQLException) {
+            throw CalibrateDataException("queryExecutionFailed")
+        }
+    }
+
+    @Suppress("SwallowedException")
+    private fun getDBConnFromPathResponse(path: String): Connection {   
+        val readOnlyProp = Properties()
+        readOnlyProp.setProperty("duckdb.read_only", "true")
+        var conn: Connection
+        try {
+            conn = DriverManager.getConnection("jdbc:duckdb:.${path}", readOnlyProp)
+        } catch (e: SQLException) {
+            throw CalibrateDataException("databaseConnectionFailed")
+        }
+        return conn
+    }
```

Comment:
This looks fragile. You should allow global exception handler to handle exception. In Kotlin, you don't need try catch  for the code to compile.

Reply:
how do we use the global exception handler to provide meaningful error messages to the user? is there somewhere you can assign a key that relates to the translation bundle? or is this not an error we have to surface to the user?

Reply:
will remove the exception for now